recipe/conda_build_config.yaml

```diff
@@ -15,9 +15,11 @@ c_stdlib:
 m2w64_c_stdlib:                # [win]
   - m2w64-toolchain            # [win]
 c_stdlib_version:              # [unix]
-  - 2.17                       # [linux]
-  - 2.17                       # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  - 2.17                       # [linux and os.environ.get("DEFAULT_LINUX_VERSION", "cos7") == "cos7"]
+  - 2.28                       # [linux and os.environ.get("DEFAULT_LINUX_VERSION", "cos7") == "alma8"]
   - 2.17                       # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  - 2.17                       # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True" and os.environ.get("DEFAULT_LINUX_VERSION", "cos7") == "cos7"]
+  - 2.28                       # [linux and os.environ.get("CF_CUDA_ENABLED", "False") == "True" and os.environ.get("DEFAULT_LINUX_VERSION", "cos7") == "alma8"]
```

Comment:
I'm pretty sure we don't want to touch the stdlib version here; for example, the newer image might just be required to satisfy the `__glibc` constraint of a test dependency, but doesn't necessarily mean that the package itself needs a newer glibc and run-constraint.

Despite the monster-zip it remains possible to override _one_ component locally - e.g. `c_stdlib_version`, and then set `os_version` for the image. I'd also prefer the choice of `c_stdlib_version` to be more visible than something implicitly implied by a knob tucked away in `conda-forge.yml`.

Reply:
> I'm pretty sure we don't want to touch the stdlib version here

OK, I've tested this hypothesis (already before commenting but) now also using a local checkout of #6626 - here's the resulting [branch](https://github.com/h-vetinari/faiss-split-feedstock/commits/pinning_test).

Basically, this works as intended if we also take the `c_stdlib_version` out of the CUDA zip, which I think is possible now. That would be the ideal scenario for me - the `c_stdlib_version` can be independently set (in CBC), from the image (in `conda-forge.yml`, or in CBC if the full zip is overridden).